GamepadButtonAndAxisEvents/explainer.md

```diff
@@ -0,0 +1,291 @@
+# Gamepad Button and Axis Events
+
+## Authors:
+
+- Sneha Agarwal (https://github.com/snehagarwal_microsoft)
+- Steve Becker (https://github.com/SteveBeckerMSFT)
+
+## Participate
+- [Issue tracker]
+- [Discussion forum]
+
+## Status of this Document
+
+This document is a starting point for engaging the community and standards bodies in developing collaborative solutions fit for standardization. As the solutions to the problems described in this document progress along the standards-track, we will retain this document as an archive and use this section to keep the community up-to-date with the most current standards venue and content location of future work and discussions.
+
+- This document status: **Active**
+- Expected venue:
+- Current version: **This document**
+
+## Introduction
+
+The current Gamepad API relies on constant polling to detect input changes, which can introduce input latency and increase CPU usage. This proposal adopts a hybrid approach that combines event-driven input handling with frame-based state consolidation, reducing unnecessary polling while maintaining responsiveness.
+
+- Event-Driven Input Handling: Immediate events (buttondown, buttonup) fire when discrete inputs like button presses or releases occur.
+
+- Frame-Based Consolidation: A gamepadchange event aggregates all input changes per frame, preventing event spam and improving efficiency.
+
+- Threshold-Based Polling for Analog Inputs: Analog inputs, such as joystick movements, are polled at a reduced frequency and only trigger updates when they exceed a defined threshold, ensuring meaningful changes are captured.
+
+By integrating these techniques, this approach minimizes CPU overhead while ensuring low-latency input processing, providing developers with a more efficient and flexible way to handle gamepad interactions. This proposal builds upon earlier work by Chromium developers while refining the hybrid model for improved performance.
+
+## User-Facing Problem
+
+The Gamepad API does not natively support event-driven input handling, and applications must rely on continuous polling for updates, which:
+
+- Introduces input latency, as polling intervals may not align with the precise timing of input changes.
+
+- Increases CPU usage, particularly when polling at high frequencies.
+
+- Lacks efficiency in handling gamepad input changes, as it does not account for meaningful input variations, leading to unnecessary events being triggered.
+
+### Goals
+
+1) **Enhance gamepad input handling** by transitioning from continuous polling to event-driven mechanisms, improving responsiveness.
+
+2) **Optimize performance** by reducing input latency and minimizing CPU overhead through a hybrid approach of event handling and threshold-based polling.
+
+3) **Provide flexibility for developers** by allowing efficient management of gamepad input events, enabling granular control over when and how input changes are processed.
```

Comment:
I am not sure that I understand what this sentence means. What "efficient management" and "granular control" the current API does not offer?